frontend/src/components/dashboard/aliases/LabelEditor.tsx

```diff
@@ -29,7 +29,8 @@ export const LabelEditor = (props: Props) => {
 
   useEffect(() => {
     setLabelJustUpdated(true);
-    setTimeout(() => setLabelJustUpdated(false), 1000);
+    const updateTimeout = setTimeout(() => setLabelJustUpdated(false), 1000);
+    return () => clearTimeout(updateTimeout);
```

Comment:
question (non-blocking): For sake of learning on my part, why do you set a timeout for this cleanup function?

Reply:
The timeout was already there, to hide the "Saved!" tooltip after a second passes. However, if the component gets unmounted before the timeout's callback executes (e.g. because the user navigates to another page within a second, or in tests), the `setLabelJustUpdated` call no longer makes sense. Thus, the change here is to drop that timeout as soon as the component unmounts.

![image](https://user-images.githubusercontent.com/4251/171601467-26d08e1b-403c-41af-94ff-e34603929f46.png)